Move `nagivateToLoginRequestUrl` out of Configuration (#7855)
AzureAD/microsoft-authentication-library-for-js: lib/msal-browser/docs/v4-migration.md

1. The `skipAuthorityMetadataCache` parameter has been removed from BrowserAuthOptions in Configuration.
1. The `protocolMode` parameter has been moved to SystemOptions instead of BrowserAuthOptions in Configuration.
1. The `supportsNestedAppAuth` parameter has been removed. Use the `createNestablePublicClientApplication` API for Nested Apps instead. Read more about Nested Apps [here](./initialization.md#nested-app-configuration).

> 1. The `navigateTologinRequestUrl` parameter has been removed from BrowserAuthOptions in Configuration and can instead now be provided on the call to `handleRedirectPromise`:

Reviewer: as a parameter to ..

Author: Updated

1. The `skipAuthorityMetadataCache` parameter has been removed from BrowserAuthOptions in Configuration.
1. The `protocolMode` parameter has been moved to SystemOptions instead of BrowserAuthOptions in Configuration.
1. The `supportsNestedAppAuth` parameter has been removed. Use the `createNestablePublicClientApplication` API for Nested Apps instead. Read more about Nested Apps [here](./initialization.md#nested-app-configuration).

> 1. The `navigateTologinRequestUrl` parameter has been removed from BrowserAuthOptions in Configuration. Default behavior remains the same (set to true), in order to maintain custom configuration:
>     1. For APIs that take a `RedirectRequest` object, the type now has a `navigateToLoginRequestUrl` parameter
>     1. For `handleRedirectPromise`, `navigateToLoginRequestUrl` has to be passed in as an option as follows:

Reviewer 2: ```suggestion
1. The `navigateTologinRequestUrl` parameter has been removed from BrowserAuthOptions in Configuration and can instead now be provided on the call to `handleRedirectPromise`:
```
